shared/typedefs/types.ts

```diff
@@ -0,0 +1,10 @@
+import { EPriority, ETaskStatus } from "./enums";
+
+export type TTask = {
+  id: number;
+  title: string;
+  description: string;
+  dueDate: Date | null;
+  priority: EPriority.HIGH | EPriority.MEDIUM | EPriority.LOW;
+  status: ETaskStatus.ACTIVE | ETaskStatus.COMPLETED;
```

Comment:
```suggestion
  priority: EPriority;
  status: ETaskStatus;
```

The ENUM itself should suffice. You don't need to set all the ENUM values individually as the type.